component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -156,7 +156,11 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
                     setResolvedUserInContext(context, authenticatedUser);
                 }
             } else if (isPreviousIdPAuthenticationFlowHandler(context)) {
-                authenticatedUser = resolveUserFromUserStore(authenticatedUser);
+                boolean isUserResolved = FrameworkUtils.getIsUserResolved(context);
+                // Resolve the user from user store if the user is not resolved in IDF handler.
+                if (!isUserResolved) {
+                    authenticatedUser = resolveUserFromUserStore(authenticatedUser);
+                }
                 setResolvedUserInContext(context, authenticatedUser);
```

Comment:
Do we need to call `setResolvedUserInContext` even if isUserResolved is true?

Reply:
yes, inside that method we are setting the step config details also. So we need that when isUserResolved is true.